src/content/getusermedia/record/js/main.js

```diff
@@ -0,0 +1,144 @@
+var playButton = document.querySelector('button#play');
+var downloadButton = document.querySelector('button#download');
+recordButton.onclick = toggleRecording;
+playButton.onclick = play;
+downloadButton.onclick = download;
+
+// window.isSecureContext could be used for Chrome
+var isSecureOrigin = location.protocol === 'https:' ||
+location.host === 'localhost';
+if (!isSecureOrigin) {
+  alert('getUserMedia() must be run from a secure origin: HTTPS or localhost.' +
+    '\n\nChanging protocol to HTTPS');
+  location.protocol = 'HTTPS';
+}
+navigator.getUserMedia(constraints, handleGumSuccess, handleGumError);
+
+function handleSourceOpen(event) {
+  console.log('MediaSource opened');
+  sourceBuffer = mediaSource.addSourceBuffer('video/webm; codecs="vp8"');
+  console.log('Source buffer: ', sourceBuffer);
+}
+
+function handleGumSuccess(stream) {
+  console.log('getUserMedia() got stream: ', stream);
+  window.stream = stream; // make available to browser console
+  if (window.URL) {
+    gumVideo.src = window.URL.createObjectURL(stream);
+  } else {
+    gumVideo.src = stream;
+  }
+}
+
+function handleGumError(error) {
+  console.log('navigator.getUserMedia error: ', error);
+}
+
+function handleDataAvailable(event) {
+  if (event.data.size > 0) {
+    recordedBlobs.push(event.data);
+    console.assert(mediaRecorder.state === 'recording',
+      'State should be "recording"');
+  } else {
+    console.assert(mediaRecorder.state === 'inactive',
+      'State should be "inactive"');
+  }
+}
+
+function handleStop(event) {
+  console.log('Recorder stopped: ', event);
+}
+
+function toggleRecording() {
+  if (recordButton.textContent === 'Start Recording') {
+    startRecording();
+  } else {
+    stopRecording();
+    recordButton.textContent = 'Start Recording';
+    playButton.disabled = false;
+    downloadButton.disabled = false;
+  }
+}
+
+function startRecording() {
+  try {
+    recordedBlobs = [];
+    mediaRecorder = new MediaRecorder(window.stream /* , 'video/vp8' */);
+  } catch (event) {
+    alert('MediaRecorder is not supported by this browser.\n\n' +
+      'Try Chrome 47 or later, with Enable experimental Web Platform features enabled from chrome://flags.');
+    console.error('Exception while creating MediaRecorder:', event);
+    return;
+  }
+  console.assert(mediaRecorder.state === 'inactive');
+  recordButton.textContent = 'Stop Recording';
+  playButton.disabled = true;
+  downloadButton.disabled = true;
+  mediaRecorder.onstop = handleStop;
+  mediaRecorder.ondataavailable = handleDataAvailable;
+  mediaRecorder.start();
+  console.log('MediaRecorder started', mediaRecorder);
+  console.assert(mediaRecorder.state === 'recording');
+}
+
+function stopRecording() {
+  mediaRecorder.stop();
+  console.log('Recorded Blobs: ', recordedBlobs);
+  // window.stream.getVideoTracks()[0].stop();
+}
+
+function play() {
+  // sourceBuffer.appendBuffer(recordedBlobs); // or...
```

Comment:
Is this supposed to be used? If not remove or add TODO: explaining why it's there.